sonar-plugin-api/src/main/java/org/sonar/api/server/rule/RulesDefinition.java

```diff
@@ -925,6 +927,21 @@ private void validate() {
       }
     }
 
+    /**
+     * Register a repository and key under which this rule used to be known
+     * (see {@link Rule#deprecatedRuleKeys} for details).
+     * <p>
+     * Deprecated keys should be added with this method in order, oldest first, for documentation purpose.
```

Comment:
IMO, this assumes something about the eventual presentation and/or order in which values will be returned by web services. It also assumes analyzers will maintain multiple deprecated keys at one time and that ... historical order is significant in any way.

Reply:
This assumes that, given user makes several calls to the method for a given rule, the question of ordering may came up, for the user's own documentation and in our futur usage of the data input by him/her through this method: "which goes first? oldest or most recent one?"

This leave no room for question by defining the order.

Whether the data is presented in WS/UI and how it is presented is irrelevant. If we know how the input is provided (here, oldest first), we can reverse if needed. What matters, is defining the contract.

Finally, if no contract is defined, then user might input A, B, C and because ordering will be left to eg. the JVM's implementation of a Set, the output won't be the same and/or will be unstable.